src/InputGroup/InputGroup.js

```diff
@@ -1,272 +1,79 @@
+import InputGroupAddon from './_InputGroupAddon';
 import PropTypes from 'prop-types';
 import withStyles from '../utils/WithStyles/WithStyles';
-import { INPUT_GROUP_ADDON_POSITIONS, INPUT_GROUP_TYPES } from '../utils/constants';
 import React, { Component } from 'react';
-
+/*eslint-disable*/
 class InputGroup extends Component {
     constructor(props) {
         super(props);
-
-        this.state = {
-            value: this.props.inputValue,
-            searchValue: this.props.inputValue
-        };
     }
 
-    handleUp = e => {
-        e.preventDefault();
-        this.setState({
-            value: parseInt(this.state.value, 10) + 1
-        });
-    };
-
-    handleDown = e => {
-        e.preventDefault();
-        this.setState({
-            value: parseInt(this.state.value, 10) - 1
-        });
-    };
-
-    handleClear = e => {
-        e.preventDefault();
-        this.setState({
-            searchValue: ''
-        });
-    };
-
-    handleChange = e => {
-        e.preventDefault();
-        this.setState({
-            searchValue: e.target.value
-        });
-    };
-
-    handleTextChange = e => {
-        e.preventDefault();
-        this.setState({
-            value: e.target.value
-        });
-    };
-
     render() {
-        const {
+        let {
             actions,
-            addonClassNames,
-            addonPos,
-            addon,
+            addon, 
             children,
             className,
             compact,
             disableStyles,
-            glyph,
-            inputClassName,
-            inputType,
-            inputId,
-            inputName,
-            inputPlaceholder,
-            inputProps,
-            inputValue,
-            localizedText,
-            numberDownButtonProps,
-            numberUpButtonProps,
             ...props
         } = this.props;
 
-        const inputGroupClasses = classnames(
-            className,
-            'fd-input-group'
-        );
-
-        const addonClasses = classnames(
-            addonClassNames,
-            'fd-input-group__addon',
-            [{ 'fd-input-group__addon--button': !!actions || inputType === 'number' }]
-        );
-
         const inputClasses = classnames(
-            inputClassName,
             'fd-input-group__input'
         );
 
-        switch (inputType) {
-            case 'number':
+        const inputGroupClasses = classnames(
+            className,
+            'fd-input-group',
+            [{ 'fd-input-group--compact': compact }]
+        );
 
-                const inputNumberClasses = classnames(
-                    inputClasses,
-                    'fd-input--no-number-spinner'
-                );
+        children = React.Children.map(children, (child) => {
+            return React.cloneElement(child, {
+                actions: actions,
+                addon: addon,
+                compact: compact,
+                className: (child.type === FormInput) ? inputClasses : '',
```

Comment:
I wouldn't match on `FormInput`.  I would match on `InputGroupAddon` and then put the `inputClasses` on anything that _didn't_ match that.  Keep in mind, the "input" here also needs to support a `FormTextarea`.